tests/bot/cogs/test_snekbox.py

```diff
@@ -21,7 +21,10 @@ async def test_post_eval(self):
         """Post the eval code to the URLs.snekbox_eval_api endpoint."""
         resp = MagicMock()
         resp.json = AsyncMock(return_value="return")
-        self.bot.http_session.post().__aenter__.return_value = resp
```

Comment:
The reason that this worked before is that when you "call" a Mock, it returns the same child mock each time. That means the lack of an argument to `post()` does not matter here: All you care about is getting the child mock that you want to manipulate the `__aenter__` return value of.

`self.bot.http_session.post()` will return the mock that will be returned later when the actual code we want to test is run and the return value of that mocks `__aenter__` mock is configured.

Reply:
I think this was written by @MarkKoz, I don't recognize it.

Reply:
Thank you for explaining. I forgot about that as it's a bit counter-intuitive to return the same thing (it is convenient for testing of course). I suppose `post.return_value.__aenter__.return_value = ...` would have worked too then, right?

Reply:
No, because the `.post` attribute is another mock than the mock it returns when you call it.

Reply:
So `post.return_value != post()`?

Reply:
I misread, you're right.